src/firebase/userCollections.js

```diff
@@ -77,5 +77,27 @@ export const removeLinkFromCollection = async (collectionName, id, user) => {
         batch.set(stats.ref, { likes: count - 1 }, { merge: true });
     }
 
-    return await batch.commit();
+    await batch.commit();
+    return link.data(); 
 };
+
+export const getUserLinksFromCollection = async(collectionName, user) => {
+    const { collectionData } = await getUserCollection(collectionName, user);
+    if (!collectionData.data()) return [];
+
+    const userLinks = getAllUserLinksFromCollection(collectionData.data().items);
+    return userLinks;
+};
+
+export const getAllUserLinksFromCollection = async(ids) => {
+    const linksRef = collection(projectFirestore, 'links');
+    const linksQuery = query(linksRef, where("__name__", "in", ids));
+    const linksDocs = await getDocs(linksQuery);
+    const parsedDocs = linksDocs.docs.map((document) => {
+        return {
+            title: document.data().title, 
+            url: document.data().url,
+        };
+    });
+    return parsedDocs || [];
+}
```

Comment:
No changes are needed for this file. See comments above.

I would move this logic to `src/hooks/useUserLikes.ts`

Reply:
Made those changes, could you take a look ?